### PR TITLE
Fix GATFinal and code for combining trees

### PR DESCRIPTION
**Pull request by jpngfile**
**Note**: I updated this PR to include pytorch ignite

I made some fixes to the `GATFinal` model and added code for combining trees together. I also created a training pipeline using pytorch ignite.

## GAT changes
- `GATFinal` was averaging across the wrong tensors, so the output was incorrect. I fixed this by changing were results are averaged.
- I changed `GloveGAT` to output to 2 nodes, since `GAT` uses softmax. I decided to do binary classification with two nodes instead of one, since the `GAT` model has an arbitrary number of output nodes and I wanted to keep the code general. It wanted, we could make the output one node by changing softmax to sigmoid. https://stats.stackexchange.com/questions/207049/neural-network-for-binary-classification-use-1-or-2-output-neurons

## Tree Processing Changes
- I added code for combing multiple trees into one large graph. Each graph can be considered a batch of trees that can be used to train the model.
- I modified the output to encode classes as one-hot vectors to fit pytorch API of calculating loss on batches of data
- I code to select only trees with at least one child to be used for training.

## Ignite training pipeline
- Added code for creating a training and validation set `create_train_and_test`
- Added code for checking how much of the data is in each binary class `get_percent_positive`. About 75% of the comments have a positive score.
- Added code to make predictions `predict_on_batch`
- Added code for training on batches
- Added code for calculating Accuracy, precision, and recall on the validation set

You can test code as follows:
- run `pip install -r requirements.txt` to install `pytorch-ignite`
- run `python process_trees.py`. This creates a pickle file of 100 processed trees
- run `cd machine_learning && python glove_gat.py`. This trains the model on the pickle file just created

### CODE REVIEW

**Comment by agpar** [comment]
Looks good!
